sequencer/sequencer.go

```diff
@@ -358,6 +359,26 @@ func (s *Sequencer) shouldSendSequences(ctx context.Context) (bool, bool) {
 	}
 
 	if err != nil {
+		// while estimating gas a new block is not created and the POE SC may return
+		// an error regarding timestamp verification, this must handled
+		if strings.Contains(err.Error(), errTimestampMustBeInsideRange) {
+			// query the sc about the value of its lastTimestamp variable
+			lastTimestamp, err := s.etherman.GetLastTimestamp()
+			if err != nil {
+				log.Errorf("failed to estimate gas for sequence batches, err: %v", err)
+				return false, false
+			}
+			// check POE SC lastTimestamp against sequences' one
+			for _, seq := range s.closedSequences {
```

Comment:
I don't understand, can't it be possible that old closed sequences have a timestamp lower than the current last timestamp from the SC?

Reply:
As the sequencer chooses timestamps, it could happen. It shouldn't, but it could. And the SC has a require that checks that, so if this ever happens, the SC won't accept the sequence.